### PR TITLE
Dev3: adding detail view controller and storing data of the recipe into parse

### PR DESCRIPTION
**Pull request by biennviquiera**
image

### CODE REVIEW

**Comment by biennviquiera** [comment]
note that a lot of these functionality is only being used for testing right now. I will be changing the event that trigger the uploading of the recipe once the swipe to save gesture is added

File: Disher/Models/Recipe.h [hunk 1]
```diff
@@ -5,12 +5,22 @@
...
 
-@interface Recipe : NSObject
+@interface Recipe : PFObject<PFSubclassing>
+
+@property (nonatomic, strong) NSString *dishName;
+@property (nonatomic, strong) NSString *imageURL;
+@property (nonatomic, strong) NSString *source;
+@property (nonatomic, strong) NSString *recipeID;
+
++ (Recipe *) initWithRecipe:(NSString *) name withURL:(NSString *) imgURL withSource:(NSString *) dishSource withID:(NSString *) recipenum;
```
Comment (melissa-gu) [inline Disher/Models/Recipe.h:20]: nit: recipeNum

File: Disher/View Controllers/RecipesViewController.m [hunk 2]
```diff
@@ -46,35 +56,53 @@ - (void)viewDidLoad {
...
+        cell.rightUtilityButtons = [self rightButtons];
+        
+    }
+    else {
+        NSString *recipeName = self.spoonResults[indexPath.row - self.mealDBresults.count][@"title"];
+        NSString *mealID = [NSString stringWithFormat:@"%@", self.spoonResults[indexPath.row - self.mealDBresults.count][@"id"]];
+        cell.recipeName.text = recipeName;
+        NSString *imageLink = self.spoonResults[indexPath.row - self.mealDBresults.count][@"image"];
+        NSURL *imageURL = [NSURL URLWithString:imageLink];
+        [cell.recipeImage setImageWithURL:imageURL];cell.recipeSource.text = @"Spoonacular";
```
Comment (melissa-gu) [inline Disher/View Controllers/RecipesViewController.m:87]: nit: separate onto 2 lines

File: Disher/View Controllers/RecipesViewController.m [hunk 3]
```diff
@@ -46,35 +56,53 @@ - (void)viewDidLoad {
...
+        NSString *mealID = self.mealDBresults[indexPath.row][@"idMeal"];
+        cell.recipeName.text = recipeName;
+        NSURL *imageURL = [NSURL URLWithString:imageLink];
+        [cell.recipeImage setImageWithURL:imageURL];
+        cell.recipeSource.text = @"TheMealDB";
+        //create uniform data model for mealdb
+        Recipe *newRecipe = [Recipe initWithRecipe:recipeName withURL:imageLink withSource:@"mealdb" withID:mealID];
+        [self.tableViewRecipes addObject:newRecipe];
+        cell.recipe = newRecipe;
+        cell.rightUtilityButtons = [self rightButtons];
```
Comment (melissa-gu) [inline Disher/View Controllers/RecipesViewController.m:78]: I don't see a corresponding line in the spoonacular logic - do we need it there? if we don't, why not?

File: Disher/View Controllers/RecipesViewController.m [hunk 4]
```diff
@@ -46,35 +56,53 @@ - (void)viewDidLoad {
...
+        NSString *recipeName = self.spoonResults[indexPath.row - self.mealDBresults.count][@"title"];
+        NSString *mealID = [NSString stringWithFormat:@"%@", self.spoonResults[indexPath.row - self.mealDBresults.count][@"id"]];
+        cell.recipeName.text = recipeName;
+        NSString *imageLink = self.spoonResults[indexPath.row - self.mealDBresults.count][@"image"];
+        NSURL *imageURL = [NSURL URLWithString:imageLink];
+        [cell.recipeImage setImageWithURL:imageURL];cell.recipeSource.text = @"Spoonacular";
+        //create uniform data model for spoonacular
+        Recipe *newRecipe = [Recipe initWithRecipe:recipeName withURL:imageLink withSource:@"spoonacular" withID:mealID];
+        [self.tableViewRecipes addObject:newRecipe];
+        cell.recipe = newRecipe;
```
Comment (melissa-gu) [inline Disher/View Controllers/RecipesViewController.m:91]: There's some shared logic here that could probably be pulled outside the if/else to reduce code duplication

File: Disher/View Controllers/RecipesViewController.m [hunk 5]
```diff
@@ -87,38 +115,140 @@ - (void) queryMealDB:(NSString *) name completionHandler:(void(^)(NSArray *retur
...
+        NSLog(@"completion");
+        dispatch_async(dispatch_get_main_queue(), ^{
+            completionHandler();
+        });
+    });
+}
+
+//Table View Cell Methods
+- (NSArray *)rightButtons
+{
```
Comment (melissa-gu) [inline Disher/View Controllers/RecipesViewController.m:189]: nit: let's keep the bracket formatting conventions the same as others in the file 😛

File: Disher/View Controllers/RecipesViewController.m [hunk 6]
```diff
@@ -87,38 +115,140 @@ - (void) queryMealDB:(NSString *) name completionHandler:(void(^)(NSArray *retur
...
+    return rightUtilityButtons;
+}
+
+- (BOOL)swipeableTableViewCellShouldHideUtilityButtonsOnSwipe:(SWTableViewCell *)cell {
+    return YES;
+}
+
+- (void)swipeableTableViewCell:(RecipeCell *)cell didTriggerRightUtilityButtonWithIndex:(NSInteger)index {
+    switch (index) {
+        case 0: { //click on save button
```
Comment (melissa-gu) [inline Disher/View Controllers/RecipesViewController.m:203]: for my understanding - what's the switch statement doing here?

File: Disher/View Controllers/RecipesViewController.m [hunk 7]
```diff
@@ -87,38 +115,140 @@ - (void) queryMealDB:(NSString *) name completionHandler:(void(^)(NSArray *retur
...
+        }];
+    });
+    dispatch_group_notify(group, dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^ {
+        // All group blocks have now completed
+        NSLog(@"completion");
+        dispatch_async(dispatch_get_main_queue(), ^{
+            completionHandler();
+        });
+    });
+}
```
Comment (melissa-gu) [inline Disher/View Controllers/RecipesViewController.m:185]: I know this was added in the last already merged [PR](https://github.com/biennviquiera/Disher/pull/6/files) - not sure if the rebase of this branch onto master worked

**Review by melissa-gu — APPROVED** [review]

File: Disher/View Controllers/RecipesViewController.m (hunk 6 above)
Comment (biennviquiera) [reply Disher/View Controllers/RecipesViewController.m:203]: the swtableviewcell uses indices for buttons so I'm using the first and only button to refer to when it is clicked

File: Disher/View Controllers/RecipesViewController.m (hunk 3 above)
Comment (biennviquiera) [reply Disher/View Controllers/RecipesViewController.m:78]: I'm not sure why it's not needed but when I ran it, both the spoonacular and mealdb cells had the same utility button being assigned to them.

**Comment by melissa-gu** [comment]
One general comment - might be helpful to have more explanation and/or videos and screenshots for the functionality added in the PR as a whole

**Comment by biennviquiera** [comment]
https://user-images.githubusercontent.com/66394477/178836059-7cc305e3-65b1-41d5-8954-08bc4352636e.mov

**Pull request merged**